pkg/grafana/synthetic-monitoring.go

```diff
@@ -28,331 +28,236 @@ import (
 	client, err := NewHttpClient()
 	if err != nil {
 		return nil, err
 	}
-	req, err := http.NewRequest("GET", url, nil)
-	req.Header.Add("Authorization", "Bearer "+authToken)
-	req.Header.Add("Content-type", "application/json")
 
-	resp, err := client.Do(req)
-	if err != nil {
-		return nil, err
-	}
-	defer resp.Body.Close()
+	const smBaseURL = "https://synthetic-monitoring-api.grafana.net"
+	smClient := smapi.NewClient(smBaseURL, "", client)
 
-	switch {
-	case resp.StatusCode == http.StatusNotFound:
-		return nil, grizzly.ErrNotFound
-	case resp.StatusCode >= 400:
-		return nil, errors.New(resp.Status)
+	apiToken, ok := os.LookupEnv("GRAFANA_SM_TOKEN")
+	if !ok {
+		return nil, fmt.Errorf("GRAFANA_SM_TOKEN environment variable must be set")
 	}
 
-	data, err := ioutil.ReadAll(resp.Body)
+	stackID, err := strconv.Atoi(os.Getenv("GRAFANA_SM_STACK_ID"))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("GRAFANA_SM_STACK_ID environment variable must be set")
 	}
-
-	var checks []Check
-	if err := json.Unmarshal(data, &checks); err != nil {
-		return nil, grizzly.APIErr{Err: err, Body: data}
-	}
-	probes, err := getProbeList()
+	metricsInstanceID, err := strconv.Atoi(os.Getenv("GRAFANA_SM_METRICS_ID"))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("GRAFANA_SM_METRICS_ID environment variable must be set")
 	}
-	for _, check := range checks {
-		if check.UID() == uid {
-			probeNames := []string{}
-			for _, probe := range check["probes"].([]interface{}) {
-				probeID := int(probe.(float64))
-				name := probes.ByID[probeID].Name
-				probeNames = append(probeNames, name)
-			}
-			check["probes"] = probeNames
-			handler := SyntheticMonitoringHandler{}
-			resource := grizzly.NewResource(handler.APIVersion(), handler.Kind(), check.Job(), check)
-			resource.SetMetadata("type", check.Type())
-			return &resource, nil
-		}
-	}
-	return nil, grizzly.ErrNotFound
-}
-
-// getRemoteCheck retrieves a check object from SM
-func getRemoteCheckList() ([]string, error) {
-	url := getSyntheticMonitoringURL("api/v1/check/list")
-	authToken, err := getAuthToken()
+	logsInstanceID, err := strconv.Atoi(os.Getenv("GRAFANA_SM_LOGS_ID"))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("GRAFANA_SM_LOGS_ID environment variable must be set")
 	}
-	client, err := NewHttpClient()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	_, err = smClient.Install(ctx, int64(stackID), int64(metricsInstanceID), int64(logsInstanceID), apiToken)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to install synthetic monitoring client : %v", err)
 	}
-	req, err := http.NewRequest("GET", url, nil)
-	req.Header.Add("Authorization", "Bearer "+authToken)
-	req.Header.Add("Content-type", "application/json")
 
-	resp, err := client.Do(req)
+	return smClient, nil
+}
+
+// getProbeList retrieves the list of probe and grouped by id and name
+func getProbeList() (*Probes, error) {
```

Comment:
No need to be pointer? You can return `Probes{}` instead.
```suggestion
func getProbeList() (Probes, error) {
```